operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -0,0 +1,722 @@
+# Copyright 2017, 2018, Oracle Corporation and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+#
+# ------------
+# Description:
+# ------------
+#
+#   This code reads the configuration in a WL domain's domain home, and generates
+#   multiple files that are copied to stdout.  It also checks whether the domain
+#   configuration is 'valid' (suitable for running in k8s).
+#
+#   This code is used by the operator to introspect and validate an arbitrary
+#   WL domain before its pods are started.  It generates topology information that's
+#   useful for running the domain, setting up its networking, and for overriding
+#   specific parts of its configuration so that it can run in k8s.
+# 
+#   The configuration overrides are specified via situational config file(s), and 
+#   include listen addresses, log file locations, etc.  Additional information
+#   is provided in other files -- including encrypted credentials, domain
+#   topology (server names, etc), and any validation warnings/errors.
+#
+#   For more details, see the Description in instrospectDomain.sh (which
+#   calls this script).
+#
+# ---------------------
+# Prerequisites/Inputs:
+# ---------------------
+#
+#   A WebLogic install.
+#
+#   A WebLogic domain.
+#
+#   A running WL node manager at port 5556 (see introspectDomain.sh).
+#
+#   Plain text WL admin username/password in:
+#     /weblogic-operator/secrets/username 
+#     /weblogic-operator/secrets/password
+#
+#   The following env vars:
+#     DOMAIN_UID  - completely unique id for this domain
+#     DOMAIN_HOME - path for the domain configuration
+#     LOG_HOME    - path to override WebLogic server log locations
+#
+# ---------------------------------
+# Outputs (files copied to stdout):
+# ---------------------------------
+#
+#   topology.yaml                  -- Domain configuration summary for operator (server names, etc).
+#                                           -and/or-
+#                                     Domain validation warnings/errors. 
+#   situational-config.xml         -- Overrides for domain configuration (listen addresses, etc).
+#   boot.properties                -- Encoded credentials for starting WL.
+#   userConfigNodeManager.secure   -- Encoded credentials for starting NM in a WL pod.
+#   userKeyNodeManager.secure'     -- Encoded credentials for starting NM in a WL pod.
+#
+# Note:
+#
+#   This code partly depends on a node manager so that we can use it to encrypt 
+#   the username and password and put them into files that can be used to connect
+#   to the node manager later in the server pods (so that the server pods don't
+#   have to mount the secret containing the username and password).
+#
+# TBD:
+#
+#   It's not clear if this script supports mixed clusters. It seems to assume
+#   a cluster with a dynamic server means the cluster is purely dynamic.
+#
```

Comment:
This TBD needs to be officially tracked.

Reply:
Done.  We will at least validate for this in 2.0, and potentially address this at some point.